Make HTTP/2-only WPT tests use the HTTP/2 server

bnham opened the pull request:
#### 7fb8f632008099b0c4e93664481ce76a02e890fa
<pre>
Make HTTP/2-only WPT tests use the HTTP/2 server
<a href="https://bugs.webkit.org/show_bug.cgi?id=253043">https://bugs.webkit.org/show_bug.cgi?id=253043</a>
rdar://problem/106003603

Reviewed by Jon Bedard.

Currently HTTP/2-only WPT tests actually run on the HTTP/1.1 server. This makes them run against the
HTTP/2 server instead as intended.

* LayoutTests/imported/w3c/web-platform-tests/fetch/api/basic/status.h2.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/basic/status.h2.any.worker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/xhr/status.h2.window-expected.txt:
* LayoutTests/platform/ios-wk2/TestExpectations:
* LayoutTests/platform/mac-wk1/TestExpectations:
* LayoutTests/platform/mac-wk2/TestExpectations:
* LayoutTests/platform/wk2/TestExpectations:
* Tools/Scripts/webkitpy/port/driver.py:
(Driver.wpt_test_path_to_uri):

Canonical link: <a href="https://commits.webkit.org/261489@main">https://commits.webkit.org/261489@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/59c5b847f354d35d27d51909a89d51547a620f5f

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/111757 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/20888 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/90/builds/372 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/87/builds/3539 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/120475 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/115817 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/22247 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/11967 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/86/builds/3280 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/117522 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/16520 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/99676 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/104719 "Built successfully") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/5/builds/115185 "Passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/98470 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/88/builds/252 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/45462 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/13359 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/89/builds/253 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/88073 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/13855 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/9686 "9 flakes 3 failures") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/19299 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/52254 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/7995 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/15841 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

gsnedders commented on Tools/Scripts/webkitpy/port/driver.py:
```diff
@@ -359,8 +359,12 @@ def is_web_platform_test(self, test_name):
     def wpt_test_path_to_uri(self, path):
-        should_use_https = ".https." in path or ".serviceworker." in path or ".serviceworker-module." in path
-        return self.web_platform_test_server_base_https_url + path if should_use_https else self.web_platform_test_server_base_http_url + path
+        if ".h2" in path:
```
```suggestion
        if ".h2." in path:
```

to be consistent with the others, I think?

bnham replied:
Yeah this is a definitely a typo on my part.

bnham committed:
Make HTTP/2-only WPT tests use the HTTP/2 server
https://bugs.webkit.org/show_bug.cgi?id=253043
rdar://problem/106003603

Reviewed by Jon Bedard.

Currently HTTP/2-only WPT tests actually run on the HTTP/1.1 server. This makes them run against the
HTTP/2 server instead as intended.

* LayoutTests/imported/w3c/web-platform-tests/fetch/api/basic/status.h2.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fetch/api/basic/status.h2.any.worker-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/xhr/status.h2.window-expected.txt:
* LayoutTests/platform/ios-wk2/TestExpectations:
* LayoutTests/platform/mac-wk1/TestExpectations:
* LayoutTests/platform/mac-wk2/TestExpectations:
* LayoutTests/platform/wk2/TestExpectations:
* Tools/Scripts/webkitpy/port/driver.py:
(Driver.wpt_test_path_to_uri):

Canonical link: https://commits.webkit.org/261489@main
